src/docdb/editors/DocDBDocumentNodeEditor.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IAzureNode } from "vscode-azureextensionui";
+import { ICosmosEditor } from "../../DocumentEditor";
+import { RetrievedDocument } from "documentdb";
+import { DocDBDocumentTreeItem } from "../tree/DocDBDocumentTreeItem";
+
+export class DocDBDocumentNodeEditor implements ICosmosEditor<RetrievedDocument> {
+    private _documentNode: IAzureNode<DocDBDocumentTreeItem>;
+    constructor(documentNode: IAzureNode<DocDBDocumentTreeItem>) {
+        this._documentNode = documentNode;
+    }
+
+    public get label(): string {
+        const collectionNode = this._documentNode.parent;
+        const databaseNode = collectionNode.parent;
+        const accountNode = databaseNode.parent;
+        return `${accountNode.treeItem.label}/${databaseNode.treeItem.label}/${collectionNode.treeItem.label}/${this._documentNode.treeItem.label}`;
+    }
+
+    public async getData(): Promise<RetrievedDocument> {
+        return this._documentNode.treeItem.document;
```

Comment:
Does these need to use Promise.resolve()?

Reply:
You don't have to for 'async' functions